controller/tests/array_action/svc/array_mediator_svc_test.py

```diff
@@ -828,31 +828,32 @@ def test_get_volume_mappings_success(self):
         mappings = self.svc.get_volume_mappings("volume")
         self.assertEqual(mappings, {'Test_P': '0', 'Test_W': '1'})
 
-    def test_get_first_free_lun_raises_host_not_found_error(self):
+    def test_get_free_lun_raises_host_not_found_error(self):
         self.svc.client.svcinfo.lshostvdiskmap.side_effect = [
             svc_errors.CommandExecutionError('Failed')]
         with self.assertRaises(array_errors.HostNotFoundError):
-            self.svc.get_first_free_lun('host')
+            self.svc.get_free_lun('host')
 
-    def test_get_first_free_lun_with_no_host_mappings(self):
+    def test_get_free_lun_with_no_host_mappings(self):
         self.svc.client.svcinfo.lshostvdiskmap.return_value = []
-        lun = self.svc.get_first_free_lun('host')
-        self.assertEqual(lun, '0')
+        lun = self.svc.get_free_lun('host')
+        self.assertTrue(self.svc.MIN_LUN_NUMBER <= int(lun) <= self.svc.MAX_LUN_NUMBER)
 
     @patch.object(SVCArrayMediator, "MAX_LUN_NUMBER", 3)
     @patch.object(SVCArrayMediator, "MIN_LUN_NUMBER", 1)
-    def test_get_first_free_lun_success(self):
+    def test_get_free_lun_success(self):
         map1 = Munch({'id': '51', 'name': 'peng', 'SCSI_id': '0',
                       'host_id': '12', 'host_name': 'Test_P'})
         map2 = Munch({'id': '56', 'name': 'peng', 'SCSI_id': '1',
                       'host_id': '16', 'host_name': 'Test_W'})
         self.svc.client.svcinfo.lshostvdiskmap.return_value = [map1, map2]
-        lun = self.svc.get_first_free_lun('Test_P')
-        self.assertEqual(lun, '2')
+        lun = self.svc.get_free_lun('Test_P')
+        self.assertTrue(self.svc.MIN_LUN_NUMBER <= int(lun) <= self.svc.MAX_LUN_NUMBER)
+        self.assertNotIn(lun, ['0', '1'])
```

Comment:
why not in this list? what is the meaning of the values in this list?

Reply:
oh, I now see them in the the maps.
could you use them with "consts" (e.g. "LUN_IN_USE_0", "LUN_IN_USE_1")

Reply:
those are the luns that are in use (part of mappings) so we are checking that its not one of them

Reply:
```suggestion
        self.assertNotIn(lun, [LUN_IN_USE_0, LUN_IN_USE_1])
```

Reply:
done